openpype/hosts/maya/plugins/publish/reset_xgen_attributes.py

```diff
@@ -24,6 +28,9 @@ def process(self, instance):
                 )
                 cmds.setAttr(node_attr, value, type="string")
             cmds.setAttr(palette + ".xgExportAsDelta", True)
-            
+
+        # Need to save the scene, cause the attribute changes above does not
+        # mark the scene as modified so user can exit without commiting the
+        # changes.
```

Comment:
We could technically also set the scene as modified explicitly `cmds.file(modified=True)` or something like that. Not sure if that's what you'd want because you could still force quit the app so likely saving is what you want. The comment here however seemed to hint that if we were able to reset the modified state we'd be ok too.

Reply:
Yes, I understand. Would be even nicer if the incremental save wouldn't end up changing the attributes to begin with. But hey, Xgen does its own. Ugh...

Reply:
Ugh indeed!